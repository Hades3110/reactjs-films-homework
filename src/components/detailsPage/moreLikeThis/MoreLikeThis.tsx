import { useEffect, useState } from 'react'
import { FilmInterface, GenreInterface, getGenres, getRecommendations } from '../../../services/filmService'
import FilmItem from '../../main/filmsList/filmItem/FilmItem'
import styles from './moreLikeThis.module.scss'
import spinner from '/public/assets/spinner.gif'

const RecFilm = ({ id }: { id: number }) => {
    const [filmArr, setFilmArr] = useState<FilmInterface[]>([])
    const [genres, setGenres] = useState<GenreInterface[]>([])
    const [isFilmFound, setIsFilmFound] = useState<boolean>(true)
    const [isLoaded, setIsLoading] = useState<boolean>(false)

    useEffect(() => {
        getRecommendations(id)
            .then(res => {
                const result = res.results.filter((film, index, self) =>
                    self.findIndex((t) => (
                        t.id === film.id
                    ))
                )
                if (result.length > 0) {
                    setFilmArr(result.filter(film => film.poster_path && film.title))
                    setIsLoading(true)
                } else {
                    setIsLoading(true)
                    setIsFilmFound(false)
                }
            })
        window.scrollTo(0, 0)
        // window.scrollTo({
        //     top: 0,
        //     behavior: "smooth"
        // })
    }, [id])

    useEffect(() => {
        getGenres()
            .then(res => setGenres(res.genres))
    }, [])

    return (<>
        <div className={styles.more}>MORE LIKE THIS</div>
        {isLoaded ?
            (isFilmFound ? <div className={styles.filmList}>
                <div className='filmListRow'>
                    {
                        filmArr.map((el: FilmInterface) => {
                            return (
                                <FilmItem
                                    key={el.id}
                                    id={el.id}
                                    title={el.title === '' ? 'NAME NOT FOUND' : el.title}
                                    image={el.poster_path}
                                    rate={el.vote_average}
                                    overview={el.overview === '' ? 'No Description' : el.overview}
                                    filmGenres={el.genre_ids}
                                    genres={genres}
                                />
                            )
                        })
                    }
                </div>
            </div> :
                <div className={styles.searchErrorMassage}>There are no movies that matched your query.</div>)
            : <img src={spinner} className={styles.spinner} />
        }
    </>)
}

export default RecFilm