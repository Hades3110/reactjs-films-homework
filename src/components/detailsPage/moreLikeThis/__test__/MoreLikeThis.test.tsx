import { create } from "react-test-renderer";
import MoreLikeThis from '../MoreLikeThis';

describe('MoreLikeThis Component', () => {

  it('MoreLikeThis snapshot test', () => {
    const component = create(<MoreLikeThis id={0} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});