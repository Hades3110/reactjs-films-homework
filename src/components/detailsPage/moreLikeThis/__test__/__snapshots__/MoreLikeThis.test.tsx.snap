// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreLikeThis Component MoreLikeThis snapshot test 1`] = `
<img
  className="spinner"
  src="test-file-stub"
/>
`;
